packaging/material/packaging_python.md

```diff
@@ -0,0 +1,180 @@
+- More on licenses in a future chapter of this course
+
+---
+
+## README.md
+
+- A `README` file typically involves detailed description about the package and also the installation procedure#
+- The `setuptools` configuration loads the `README.md` file to get the `long_description`, hence it is mandatory to include this file
+- Newer versions of `setuptools` also include this file automatically
+
+---
+
+## Creating distribution archives 1/2
+
+- We need to convert the folder sturcture that we have into an archive file system which the end user can download and install on their system
+- To do this we will use the package builder [build](https://pypa-build.readthedocs.io/en/stable/index.html) which was introduced in [PEP 517](https://www.python.org/dev/peps/pep-0517/)
+- These archive files are finally uploaded to PyPI and can be installed by pip
+- The disctribution archives are generated by running the command `python3 setup.py sdist` in the `package_name/` repository
+- The following directory and files will be generated in the repository:
+
+```python
+dist/
+  package_name_YOUR_USERNAME_HERE-0.0.1-py3-none-any.whl
+  package_name_YOUR_USERNAME_HERE-0.0.1.tar.gz
+```
+
+---
+
+## Uploading the distribution archives 1/2
+
+- Once the distribution archives are created, we want to upload it to the Python Package Index
+- We will use [Twine](https://twine.readthedocs.io/en/latest/) to upload our package to PyPI
+    - Twine is widely used as it securely authenticates the user to PyPI over HTTPS using a verified connection
+    - Its predecessor `python setup.py upload` required careful configuration of the user system and Python version
+- PyPI offers an alternative index for testing called TestPyPI on which a user can upload and experiment with their package
+- It is always a good idea to verify uploading over TestPyPI before uploading to the real PyPI
+
+---
+
+## Uploading the distribution archives 2/2
+
+- Before using TestPyPI one needs an account and a PyPI API Token
+- Account generation is straightforward through a [registration procedure](https://test.pypi.org/account/register/)
+- API Token generation is also [a similar procedure](https://test.pypi.org/account/login/?next=%2Fmanage%2Faccount%2F#api-tokens)
+- The token `__token__` is the username and the token value is the password required for uploading
+- Uploading to TestPyPI can be done using the command: `python3 -m twine upload --repository testpypi dist/*`
+- The upload looks something like:
+
+```python
+Uploading distributions to https://test.pypi.org/legacy/
+Enter your username: [your username]
+Enter your password:
+Uploading package_name_YOUR_USERNAME_HERE-0.0.1-py3-none-any.whl
+100%|█████████████████████| 4.65k/4.65k [00:01<00:00, 2.88kB/s]
+Uploading package_name_YOUR_USERNAME_HERE-0.0.1.tar.gz
+100%|█████████████████████| 4.25k/4.25k [00:01<00:00, 3.05kB/s]
+```
+
+---
+
+## Installing the uploaded package
```

Comment:
This end feels a bit sudden. Maybe some summary slide, slide about further important but less commonly used features (just to name them and link to them)?!